qualtran/surface_code/ccz2t_cost_model.py

```diff
@@ -204,11 +225,60 @@ def get_ccz2t_costs(
         data_d = code_distance_from_budget(phys_err=phys_err, budget=target_err_per_round)
         data_block = SimpleDataBlock(data_d=data_d, routing_overhead=routing_overhead)
 
-    data_error = data_block.data_error(
-        n_algo_qubits=n_algo_qubits, n_cycles=n_cycles, phys_err=phys_err
+    return get_ccz2t_costs(
+        n_magic=n_magic,
+        n_algo_qubits=n_algo_qubits,
+        phys_err=phys_err,
+        cycle_time_us=cycle_time_us,
+        factory=factory,
+        data_block=data_block,
     )
-    failure_prob = distillation_error + data_error
-    footprint = factory.footprint() + data_block.footprint(n_algo_qubits=n_algo_qubits)
-    duration_hr = (cycle_time_us * n_cycles) / (1_000_000 * 60 * 60)
 
-    return PhysicalCost(failure_prob=failure_prob, footprint=footprint, duration_hr=duration_hr)
+
+def iter_ccz2t_factories(
+    l1_start: int = 5, l1_stop: int = 25, l2_stop: int = 41
+) -> Iterator[CCZ2TFactory]:
+    for l1_distance in range(l1_start, l1_stop, 2):
+        for l2_distance in range(l1_distance + 2, l2_stop, 2):
+            yield CCZ2TFactory(distillation_l1_d=l1_distance, distillation_l2_d=l2_distance)
+
+
+def iter_simple_data_blocks(d_start: int = 7, d_stop: int = 35):
+    for logical_data_qubit_distance in range(d_start, d_stop, 2):
+        yield SimpleDataBlock(data_d=logical_data_qubit_distance)
+
+
+def get_ccz2t_costs_from_grid_search(
+    *,
+    n_magic: AlgorithmSummary,
+    n_algo_qubits: int,
+    phys_err: float = 1e-3,
+    cycle_time_us: float = 1.0,
+    factory_iter: Callable[[], Iterator] = iter_ccz2t_factories,
+    data_block_iter: Callable[[], Iterator] = iter_simple_data_blocks,
+) -> Tuple[PhysicalCost, CCZ2TFactory, SimpleDataBlock]:
+    """Grid search over parameters to minimize space time volume.
+
+    This is from openfermion. What paper was it introduced or used in?
```

Comment:
As far as I understand, this search was performed by hand using a tweaked version of the CCZ2T spreadsheet in the [costing of THC (page 25)](https://link.aps.org/doi/10.1103/PRXQuantum.2.030305).

Reply:
Ah great. Kindly add that to the docstring